### ISSUE TITLE
ReceivedNotificationHandler does not get called when using Large Icon or Big Picture

### ISSUE THREAD

**Issue by anderscheow** [OneSignal/OneSignal-Flutter-SDK#103]
In Android, whenever I add `large_icon`, `big_picture` or both in the create notification API, `ReceivedNotificationHandler` does not get called. It works when I removed both `large_icon` and `big_picture`.

**Comment by rgomezp** [comment]
Howdy,
Please use the OneSignal `setLogLevel` method and paste your logs below. Thanks!

**Comment by anderscheow** [comment]
Log
```
D/onesignal(15789): Created json raw payload: {vis=1, google.original_priority=normal, google.sent_time=1563521451477, licon=https://jom-invites.s3.amazonaws.com/banners/c3878790-a51b-11e9-b2dd-f340076d12f7.octet-stream, google.delivered_priority=normal, custom={"a":{"pl_title":"Zuerrrrrrr responded to your invitation","invitationId":"5d1b41d27e7f5e6d34e89b9d","pl_body":"Zuerrrrrrr is going to Tomorrowland 2021 with 11 headcount.","pl_type":"RESPOND_INVITATION","memberId":"5cf291e7f83c8e106f91b5a5","status":"GOING"},"i":"cb7e196e-ebe8-4004-a290-2ec2ba3e9339"}, title=Zuerrrrrrr responded to your invitation, chnl={"dscr":"Anything related to invitation.","grp_nm":"Invitation","grp_id":"OS_2daa36d6-3a08-46cd-bbad-7b7050c9d32b","id":"OS_17b568df-3182-473b-a7bf-30f3b5ce2ccb","nm":"Invitation"}, google.message_id=0:1563521451484876%9c5766ecf9fd7ecd, alert=Zuerrrrrrr is going to Tomorrowland 2021 with 11 headcount., google.ttl=259200, from=762914440315, notificationId=1601362240}
D/FA      (15789): Logging event (FE): os_notification_received, Bundle[{notification_id=cb7e196e-ebe8-4004-a290-2ec2ba3e9339, medium=notification, source=OneSignal, campaign=Zuerrrrrrr, firebase_event_origin(_o)=app, firebase_screen_class(_sc)=Flutter, firebase_screen_id(_si)=5029601666557752082, firebase_screen(_sn)=/home}]
V/FA      (15789): Connecting to remote service
D/FA      (15789): Connected to remote service
V/FA      (15789): Processing queued up service tasks: 1
V/FA      (15789): Inactivity, disconnecting from the service
```

Code
```Dart
void _configureOneSignal() {
    OneSignal.shared.setLogLevel(OSLogLevel.verbose, OSLogLevel.none);

    var settings = {
      OSiOSSettings.autoPrompt: true,
      OSiOSSettings.promptBeforeOpeningPushUrl: true
    };

    OneSignal.shared.setSubscriptionObserver((OSSubscriptionStateChanges changes) {
      print("SUBSCRIPTION STATE CHANGED: ${changes.jsonRepresentation()}");
      this._savePushToken(changes);
    });

    OneSignal.shared.setNotificationReceivedHandler((notification) {
      print("Received notification: \n${notification.payload.additionalData.toString()}");
      this._identifyCategoryFromPayload(notification.payload.additionalData);
    });

    OneSignal.shared.setNotificationOpenedHandler((OSNotificationOpenedResult result) {
      print("Opened notification: \n${result.notification.jsonRepresentation().replaceAll("\\n", "\n")}");
      this._analyzeAndPublishPayload(result.notification.payload.additionalData);
    });

    OneSignal.shared.setLocationShared(true);

    OneSignal.shared.promptLocationPermission();

    OneSignal.shared.setInFocusDisplayType(OSNotificationDisplayType.notification);

    OneSignal.shared.init(this.oneSignalAppId, iOSSettings: settings);

    OneSignal.shared.consentGranted(true);
  }
```

**Comment by jkasten2** [comment]
@anderscheow Do you still get the notification in the notification shade?

There is a different flow when a remote file has to be loaded to generate the notification, a job service has to be created since a broadcast receiver might not be able to download the file in time. This causing the callback to fire from a different thread.

We just updated all our callbacks to run on the main thread so I believe this issue should be fixed in the 2.0.1 release.

Please follow our 2.0.0 migration guide to upgrade to this new plugin.
https://github.com/OneSignal/OneSignal-Flutter-SDK/releases/tag/2.0.0

**Comment by anderscheow** [comment]
@jkasten2 Once I updated to 2.0.1 release, it works just as it expected. Thanks for the hard work and respond!
